lib/buildsystems/qmake.rb

```diff
@@ -4,7 +4,7 @@ class Qmake < Package
   property :qmake_build_extras, :qmake_install_extras
 
   def self.build
-    system "QMAKE_CXX='g++ #{ARCH == 'x86_64' && LIBC_VERSION.to_f >= 2.35 ? File.join(CREW_LIB_PREFIX, 'libC.so.6').to_s : ''}' qmake"
+    system "QMAKE_CXX='g++ #{ARCH == 'x86_64' && Gem::Version.new(LIBC_VERSION.to_s) >= Gem::Version.new('2.35') ? File.join(CREW_LIB_PREFIX, 'libC.so.6').to_s : ''}' qmake"
```

Comment:
Why this change? Isn't it just a more complicated way of doing the same thing?

Reply:
There was a discussion earlier of version comparisons possibly breaking in the glibc builds, so as per that I've been unifying the usage with the gem version comparisons.

Reply:
Yes, when comparing versions 2.4 is lower than 2.37, however 2.40 is higher than 2.37.  That is where the potential problem may occur.

Reply:
Yes, but the lowest glibc version we will encounter is 2.23, so we don't need to worry about that. All the glibc versions we deal with have double digit minor versions.

Reply:
Honestly I don't care which way we go on this. I just want it to be uniform if possible. I think that glibc versions are a special case, and we should avoid special cases when possible...

Reply:
Let's unify it to just the float comparison-- `Gem::Version.new` is overkill here.

Reply:
I just explained why we need to compare versions semantically a few comments earlier.

Reply:
Bedtime reading material: https://semver.org/

Reply:
> I just explained why we need to compare versions semantically a few comments earlier.

And I explained a few comments later why that is not relevant in the case of comparing glibc versions.

glibc version 2.4 was released in 2005, 5 years before the first chromebook released. and again, the lowest glibc version we will encounter is 2.23.

Reply:
Remember, versions can be of the form 2.37.1.  So, if LIBC_VERSION is converted to float in this instance, the value would be 2.37 which compared to an earlier version may or may not yield the desired result.  We cannot assume glibc will always have one decimal point.  A lot of the earlier glibc versions had 2 decimal points so we better use the proper version comparison.